### PR TITLE
Add a warning for databinding to undeclared properties/attributes

### LINKED ISSUE #25: Binding to undefined attribute/property
Should be ported from polylint, see lintBindingExpression here: https://github.com/PolymerLabs/polylint/blob/master/src/linters.ts#L299

### PR DESCRIPTION
**Pull request by rictic**
- [x] CHANGELOG.md has been updated

Fixes #25

---
This change is [Reviewable](https://reviewable.io/reviews/polymer/polymer-linter/35)

### CODE REVIEW

File: src/polymer/bind-to-undeclared.ts [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+  }
+}
+
+function closestOption(name: string, isAttribute: boolean, element: Element) {
+  const attributeOptions = element.attributes.map((a) => a.name)
+                               .concat(Array.from(sharedAttributes.keys()));
+  const propertyOptions = element.properties.map((a) => a.name)
+                              .concat(Array.from(sharedProperties.keys()));
+  const closestAttribute =
+      minBy(attributeOptions, (option) => levenshtein.get(name, option));
```
Comment (usergenic) [inline src/polymer/bind-to-undeclared.ts]: Cool!

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,17 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
...
 
+### New Lint Rules
+- `unknown-set-attribute` - included in all polymer rule collections.
+  - Warns setting undeclared properties or attributes in Polymer templates.
+
+    This rule will warn when setting an attribute or property on a custom
+    element.
+
+    This catches misspellings, forgetting to convert camelCase to kebab-case,
+    and binding to attributes like class and style like they were properties.Warns when setting an unknown attribute or property on a custom element in a polymer databinding template. Catches misspellings
```
Comment (usergenic) [inline CHANGELOG.md]: `properties.\n    Warns`

File: src/polymer/bind-to-undeclared.ts [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+  'class-name',
+  'slot',
+
+
+  'is',
+]);
+
+
+export class BindToUndeclaredAttributes extends HtmlRule {
+  code = 'unknown-set-attribute';
```
Comment (usergenic) [inline src/polymer/bind-to-undeclared.ts]: Why is this code so different from the rule class name?

Comment (rictic) [reply src/polymer/bind-to-undeclared.ts]: Good catch, updated

File: CHANGELOG.md (hunk 2 above)
Comment (rictic) [reply CHANGELOG.md]: Done

File: src/polymer/bind-to-undeclared.ts (hunk 1 above)
Comment (rictic) [reply src/polymer/bind-to-undeclared.ts]: :D

**Comment by rictic** [comment]
Comments addressed. Also now understands and correctly lints setting attributes outside a polymer databinding context too.

**Review by usergenic — APPROVED** [review]
LGTM

**Pull request merged**